### PR TITLE
[CSH-44] Create Teaching Staff Profile Frontend

### PR DESCRIPTION
**Pull request by Ali-Mones**
profile
edit

### CODE REVIEW

**Review by MarwanElsays — APPROVED** [review]
Very coool design, i like ittt🤩🤩♥️

File: React/central-student-hub/src/UserProfileComponent/Models/OfficeHours.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type OfficeHours = {
```
Comment (MuhammedAdelTaha) [inline React/central-student-hub/src/UserProfileComponent/Models/OfficeHours.ts:1]: Rename this to be OfficeHour.

File: React/central-student-hub/src/UserProfileComponent/Models/TeachingStaffProfileInfo.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { OfficeHours } from "./OfficeHours"
```
Comment (MuhammedAdelTaha) [inline React/central-student-hub/src/UserProfileComponent/Models/TeachingStaffProfileInfo.ts:1]: Rename this to be OfficeHour.

File: React/central-student-hub/src/UserProfileComponent/Models/TeachingStaffProfileInfo.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export type TeachingStaffProfileInfo = {
+    teacherId: number,
+    firstName: string,
+    lastName: string,
+    biography: string,
+    profilePictureUrl: string,
+    department: string
+    contactData: ContactInfo[],
+    officeHours: OfficeHours[]
```
Comment (MuhammedAdelTaha) [inline React/central-student-hub/src/UserProfileComponent/Models/TeachingStaffProfileInfo.ts:12]: Rename this to be OfficeHour[].

File: React/central-student-hub/src/UserProfileComponent/Models/OfficeHours.ts (hunk 1 above)
Comment (Ali-Mones) [reply React/central-student-hub/src/UserProfileComponent/Models/OfficeHours.ts:1]: There are multiple hours in a single instance of this type, so I believe this naming is better!

Comment (MuhammedAdelTaha) [reply React/central-student-hub/src/UserProfileComponent/Models/OfficeHours.ts:1]: Yes, you are right. I didn't notice.

**Review by MuhammedAdelTaha — APPROVED** [review]

**Pull request merged**